test/Test/Toml/Parsing/Unit.hs

```diff
@@ -293,6 +319,18 @@ spec_Parser = do
       --  stringFailOn (squote "\nabc")
       --  stringFailOn (squote "ab\r\nc")
       --  stringFailOn (squote "abc\n")
+    context "when the string is a multi-line literal string" $ do
+      let squote3 = quoteWith "'''"
+
+      it "can parse multi-line strings surrounded by three single quotes" $ do
+        parseText (squote3 "first line \nsecond.\n   3\n") "first line \nsecond.\n   3\n"
```

Comment:
Let's use [`neat-interpolation`](http://hackage.haskell.org/package/neat-interpolation) package for testing. So instead of writing string like this:

```haskell
squote3 "first line \nsecond.\n   3\n"
```

You will just write like this (using `{-# LANGUAGE QuasiQuotes #-}`:

```haskell
[text|
'''first line
second.
    3
'''
|]
```

This makes it more visible what actually parsed. Tests will become bigger, I understand. But we will think how to split tests later.

Reply:
Instead of `parseText (squote3 "first line \nsecond.\n   3\n") "first line \nsecond.\n   3\n"`, I've tried to do this:
```haskell
parseText
[text|
'''first line
second.
    3
'''
|]
[text|
first line
second.
   3

|]
```
Now I'm getting a long error message but it starts with:
```
 • Couldn't match expected type ‘Data.Text.Internal.Text -> b0’
               with actual type ‘Data.Text.Internal.Text’
```
I'm gonna have to figure out this error before making the other changes. Any ideas about the cause of the error?

Reply:
@ghallak Could you provide complete error message with actual code? Did you enable `QuasiQuotes`? Maybe you need to add extra spaces to arguments, not start them from beginning of line, like this:

```haskell
parseText
    [text|
    ...
    |]
    [text|
    ...
    |]
```

Reply:
Adding extra spaces to the arguments fixed the problem, but now I notice that:
```haskell
[text|
1
|]
```
is the same as:
```haskell
[text|
1


|]
```
so it's ignoring trailing newlines, and that will make it not possible to rewrite some of the tests.

Also, is there a way to insert a control character like `\b` for example in a `[text| ... |]` block? that's also required to rewrite some of the tests.

Reply:
@ghallak Ah, shit, `neat-interploation` actually doesn't fit here... Okay, let's forget this idea and keep tests as it is now. I will think of better solution for this problem later. There's a workaround for trailing empty lines. But I dunno about control characters...

Reply:
What's the workaround for trailing empty lines? :thinking:

Reply:
The idea is to have variable like `endLine = "\n"` (empty string probably will work as well) and then write like this:

```haskell
[text|


$endLine
|]
```

Also works if needed to keep indentation.